services/libs/integrations/src/integrations/reddit/processStream.ts

```diff
@@ -0,0 +1,186 @@
+import { ProcessStreamHandler, IProcessStreamContext } from '../../types'
+import {
+  RedditStreamType,
+  IRedditSubredditStreamData,
+  RedditPostsResponse,
+  IRedditCommentStreamData,
+  IRedditMoreCommentsStreamData,
+  RedditCommentsResponse,
+  RedditComment,
+  RedditMoreChildren,
+  RedditMoreCommentsResponse,
+  IRedditPublishData,
+  RedditActivityType,
+} from './types'
+import getPosts from './api/getPosts'
+import getComments from './api/getComments'
+import getMoreComments from './api/getMoreComments'
+
+async function recursiveCommentParser(
+  kind: string,
+  comment: RedditComment | RedditMoreChildren,
+  sourceParentId: string,
+  ctx: IProcessStreamContext,
+) {
+  const metadata = ctx.stream.data as IRedditCommentStreamData
+
+  // If the kind is 'more', instead of a comment we have a list of comment IDs to expand. We need to create streams for those and return them.
+  if (kind === 'more') {
+    comment = comment as RedditMoreChildren
+    ctx.log.debug(
+      { stream: ctx.stream.identifier, childrenLength: comment.children.length },
+      'Found more children to parse',
+    )
+    // Split list into chunks of 99
+    // eslint-disable-next-line no-inner-declarations
+    function* chunks<T>(arr: T[], n: number): Generator<T[], void> {
```

Comment:
you have `partition` method in the `@crowd/common`. It's not a generator but still.